django/db/backends/postgresql/base.py

```diff
@@ -373,12 +375,14 @@ def _configure_connection(self, connection):
         # variables.
 
         # Commit after setting the time zone.
-        commit_tz = ensure_timezone(connection, self.ops, self.timezone_name)
+        commit_tz = self.init_connection_timezone(
```

Comment:
So `_configure_connection` still takes self and passes `self.ops` in, so I guess if we use the pool we still hold a reference to the outer `DatabaseConnection` object. Could this cause us problems down the road? I do not really see many ways to salvage this. Either we make `_configure_connection` and everything called by it completely free of `self` or none of it`. I we don't get rid of `self` we still have this weird situation that the pool as a reference to one arbitrary connection from Django. I wonder if it would make sense to instantiate a new connection in `def pool(self)` which is not used by Django and let the pool keep it's reference on that. What are your thoughts?

Reply:
I kept missing the point but I get what you're saying now - yes I agree that situation is weird 
I have tried again :pray: